go/processor/floria/precompiled_test.go

```diff
@@ -80,3 +81,347 @@ func TestPrecompiled_AddressesAreHandledCorrectly(t *testing.T) {
 		})
 	}
 }
+
+func TestPrecompiled_StateSetBalance(t *testing.T) {
+	tests := map[string]struct {
+		input  []byte
+		gas    tosca.Gas
+		sender tosca.Address
+		err    error
+	}{
+		"outOfGas": {
+			make([]byte, 64),
+			1,
+			tosca.Address{},
+			ErrOutOfGas,
+		},
+		"invalidInput": {
+			make([]byte, 63),
+			9000,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"sameAddress": {
+			make([]byte, 64),
+			9000,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"successful": {
+			make([]byte, 64),
+			9000,
+			tosca.Address{0x01},
+			nil,
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+			if test.err == nil {
+				state.EXPECT().SetBalance(tosca.Address{}, tosca.Value{})
+			}
+
+			gas, err := stateSetBalance(state, test.sender, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != 0 {
+				t.Errorf("unexpected gas, want %v, got %v", 0, gas)
+			}
+		})
+	}
+}
+
+func TestPrecompiled_StateCopyCode(t *testing.T) {
+	codeSize := tosca.Gas(42)
```

Comment:
I guess a `codeSize` should not be of type `tosca.Gas`, right? (also in other tests)